src/storable.rs

```diff
@@ -492,6 +492,102 @@ where
     };
 }
 
+fn serialize_bounded_with_size<T>(entry: &T, bytes: &mut [u8], start: usize) -> usize
+where
+    T: Storable,
+{
+    let bounds = bounds::<T>();
+    let size_len = bytes_to_store_size(&bounds) as usize;
+    let max_size = bounds.max_size as usize;
+    let entry_bytes = entry.to_bytes();
+    debug_assert!(entry_bytes.len() <= max_size);
+
+    encode_size(
+        &mut bytes[start..start + size_len],
+        entry_bytes.len(),
+        &bounds,
+    );
+
+    bytes[start + size_len..start + size_len + entry_bytes.len()]
+        .copy_from_slice(entry_bytes.borrow());
+
+    start + max_size + size_len
+}
+
+fn deserialize_bounded_with_size<T>(bytes: &[u8], start: usize) -> (T, usize)
+where
+    T: Storable,
+{
+    let bounds = bounds::<T>();
+    let max_size = bounds.max_size as usize;
+    let size_len = bytes_to_store_size(&bounds) as usize;
+    let len = decode_size(&bytes[start..start + size_len], &bounds);
+
+    let a = T::from_bytes(Cow::Borrowed(
+        &bytes[start + size_len..start + size_len + len],
+    ));
+    (a, start + max_size + size_len)
+}
+
+impl<A, B, C> Storable for (A, B, C)
+where
+    A: Storable,
+    B: Storable,
+    C: Storable,
+{
+    fn to_bytes(&self) -> Cow<[u8]> {
+        match Self::BOUND {
+            Bound::Bounded { max_size, .. } => {
+                let mut bytes = vec![0; max_size as usize];
+
+                let a_end = serialize_bounded_with_size(self.0.borrow(), &mut bytes, 0);
+                let b_end = serialize_bounded_with_size(self.1.borrow(), &mut bytes, a_end);
+                serialize_bounded_with_size(self.2.borrow(), &mut bytes, b_end);
+                Cow::Owned(bytes)
+            }
+            _ => todo!("Serializing tuples with unbounded types is not yet supported."),
+        }
+    }
+
+    fn from_bytes(bytes: Cow<[u8]>) -> Self {
+        match Self::BOUND {
+            Bound::Bounded { max_size, .. } => {
+                assert_eq!(bytes.len(), max_size as usize);
+
+                let (a, a_end) = deserialize_bounded_with_size::<A>(bytes.borrow(), 0);
+                let (b, b_end) = deserialize_bounded_with_size::<B>(bytes.borrow(), a_end);
+                let (c, _) = deserialize_bounded_with_size::<C>(bytes.borrow(), b_end);
+
+                (a, b, c)
+            }
+            _ => todo!("Deserializing tuples with unbounded types is not yet supported."),
+        }
+    }
+
+    const BOUND: Bound = {
+        match (A::BOUND, B::BOUND, C::BOUND) {
+            (Bound::Bounded { .. }, Bound::Bounded { .. }, Bound::Bounded { .. }) => {
+                let a_bounds = bounds::<A>();
+                let b_bounds = bounds::<B>();
+                let c_bounds = bounds::<C>();
+
+                Bound::Bounded {
+                    max_size: a_bounds.max_size
```

Comment:
Do we need to actually store the sizes if all the elements are fixed? I think we can get away with not storing them at all in that case to save some bytes.

Reply:
Do you think that we do not store `Bound` for `tuples (A, B, C)` at all?

Reply:
If that is your idea I agree, I can do that.

Reply:
I'm not sure I understand your question, but here's an example: suppose you're storing `(u8, u8, u8)`. `u8` is a fixed type, in the sense its size is _always_ one byte. In that case, we don't need to store the size in the tuple encoding because we already know it.

Reply:
Done, PTAL.